CMakeLists.txt

```diff
@@ -101,7 +101,10 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Os -Wno-sized-deallocation -fno-sized-d
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DDEBUG -DSANITIZE_PTRS")
 add_link_options(--sysroot ${CMAKE_BINARY_DIR}/Root)
 
-include_directories(Libraries/LibC)
+include_directories(SYSTEM Libraries/LibC)
+include_directories(SYSTEM Libraries/LibM)
+include_directories(SYSTEM Libraries/LibPthread)
+
```

Comment:
What do we need this for? `serenity_libc()` should already add the headers to the path of whoever depends on those libraries. Maybe change `serenity_libc()` to use `SYSTEM` instead?

Reply:
Ah, i guess the idea was that when building an app cross-compiled or within serenity, the system headers in LibC, LibM, and LibPthread should be `-isystem`. But, it's true that if you're linking against these libs you'll get the headers, i overlooked that.

Reply:
Yeah, including them with `-isystem` is probably a good idea (although it shouldn't matter for us). But maybe you can do that with the existing `target_include_directories()` command in `serenity_libc()`??

Reply:
There's no `target_include_directories()` directive in the `serenity_libc()` function? Only found references to that in LibM/CMakeLists.txt and LibPthread/CmakeLists.txt, where it's marked PUBLIC.

I can change those to SYSTEM and add one to LibC/CMakeLists.txt? Haven't used cmake much before but that sounds like a reasonable change.

Reply:
Right, I was thinking of `target_link_directories()`. Maybe add `target_include_directories()` there then and remove it from individual libcs?

Reply:
See https://github.com/SerenityOS/serenity/pull/2242/commits/fa7038d67b8c3da089dde9ed6a316d663346816e. I may have gone a bit overboard...